### PR TITLE
fix(sourcemaps): Prevent Debug ID collisions

### LINKED ISSUE #3350: Sentry CLI mapping identical empty source maps to same debug ID
### Environment

- `@sentry/cli` 2.58.6 (deterministic debug IDs, i.e. since #1570; presumably affects all recent versions)
- Invoked as `sentry-cli sourcemaps inject <dir>` then `sentry-cli sourcemaps upload <dir>`
- Build: esbuild-based (Angular `@angular/build:application` v21), code splitting on, `sourceMap: true`
- OS: Linux, Docker `node:22-bullseye-slim`

## Steps to Reproduce

1. Produce a build where two or more **distinct** minified JS chunks have **byte-identical** source maps. This happens naturally with esbuild-generated chunks that have no original source — generator/async runtime helpers (`function*` / `__await` / `__generator`) and thin re-export shims — whose maps are empty:

   ```json
   {"version":3,"sources":[],"names":[],"mappings":""}
   ```

2. Run `sentry-cli sourcemaps inject dist`.
3. Inspect the injected IDs: `grep -rhoE '//# debugId=[0-9a-f-]+' dist | sort | uniq -d`.

   Minimal repro sketch (two different JS files, identical empty maps):

   ```bash
   mkdir -p dist
   printf 'console.log(1)\n//# sourceMappingURL=a.js.map\n' > dist/a.js
   printf 'console.log(22)\n//# sourceMappingURL=b.js.map\n' > dist/b.js
   printf '{"version":3,"sources":[],"names":[],"mappings":""}' > dist/a.js.map
   printf '{"version":3,"sources":[],"names":[],"mappings":""}' > dist/b.js.map
   npx -p @sentry/cli sentry-cli sourcemaps inject dist
   grep -H debugId dist/*.js
   ```

### Expected Result

Each distinct minified file gets a distinct debug ID (debug IDs identify the *minified artifact*, so two different artifacts must not share one).

### Actual Result

All chunks whose source maps are identical receive the **same** debug ID, because the deterministic ID is derived from the source-map content and empty maps are byte-identical.

Real-world observation:

Image

There are three sources all mapped to the same debug ID. Source files are different (see the different file size), sourcemap files are identical:

```json
{"version":3,"sources":[],"names":[],"mappings":"","debugId":"45eac7fa-f181-5cca-93b7-cfb58909dbb3"}
```

At event processing this collision makes Sentry resolve a frame from one chunk against a *different* chunk's artifact, and emit `js_invalid_sourcemap_location` — frames in `chunk-DZYMZJAH.js` show the source context of `chunk-S6YTOZHQ.js`, which is very confusing.

### Suggested fix

Derive the deterministic debug ID from the **minified source file** (which differs across these chunks) rather than from the source map (identical when empty), or otherwise guarantee uniqueness per minified artifact — so chunks with empty/identical maps don't collide onto one debug ID.

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
Generate Debug IDs from both the generated JS bytes and the sourcemap bytes when a sourcemap is available. This keeps distinct JS files from sharing a Debug ID when their maps are byte-identical, while preserving existing IDs already present in JS or sourcemaps.

Add a regression test for identical empty maps with distinct JS inputs. Update the sourcemaps inject snapshots because the injected Debug IDs and embedded source map payloads now change for every affected case.

Fixes #3350
Fixes [CLI-341](https://linear.app/getsentry/issue/CLI-341/sentry-cli-mapping-identical-empty-source-maps-to-same-debug-id)

### CODE REVIEW

**Comment by szokeasaurusrex** [comment]
@JPeer264 this change is needed to fix #3350, but it also will change all the debug IDs (i.e. the same sourcemap/minified file pair in the previous version will have a different debug ID after this update is released).

I am assuming that this is okay, but are you aware of any situation where this could break the product?

**Review by loewenheim — APPROVED** [review]

**Review by JPeer264 — APPROVED** [review]
That should be alright

File: src/utils/sourcemaps.rs [hunk 1]
```diff
@@ -896,9 +906,14 @@ impl SourceMapProcessor {
...
-                                        let debug_id = inject::debug_id_from_bytes_hashed(
-                                            &sourcemap_file.contents,
+                                        let source_file = self.sources.get(source_url).expect(
+                                            "source file must exist when processing its sourcemap",
                                         );
+                                        let debug_id =
+                                            inject::debug_id_from_js_and_sourcemap_bytes_hashed(
+                                                source_file.contents.as_slice(),
+                                                sourcemap_file.contents.as_slice(),
+                                            );
```
[the remark this answers is not in the document]
Comment (szokeasaurusrex) [reply src/utils/sourcemaps.rs:916]: This is probably not a very common occurrence in the real world. We are therefore choosing not to address this unless we receive a bug report
